build_tools/azure/test_script.sh

```diff
@@ -29,6 +29,18 @@ if [[ "$BUILD_REASON" == "Schedule" ]]; then
     export SKLEARN_RUN_FLOAT32_TESTS=1
 fi
 
+# debug
+echo full env
+env
```

Comment:
This feels fairly noisy when trying to look at test failures [on the ci](https://dev.azure.com/scikit-learn/scikit-learn/_build/results?buildId=45695&view=logs&j=78a0bf4f-79e5-5387-94ec-13e67d216d6e&t=f1857171-4a53-55c7-3ab5-90acfe091baa). Do you intend to have the full env displayed here all the time?